Source/Core/Common/Misc.cpp

```diff
@@ -8,27 +8,19 @@
 
 #include "Common/CommonFuncs.h"
 
-// Neither Android nor OS X support TLS
-#if  defined(__APPLE__) || (ANDROID && __clang__)
-#define __thread
-#endif
-
 // Generic function to get last error message.
 // Call directly after the command or use the error num.
 // This function might change the error code.
-const char* GetLastErrorMsg()
+std::string GetLastErrorMsg()
 {
-	static const size_t buff_size = 255;
+	const size_t buff_size = 255;
+	char err_str[buff_size] = {};
```

Comment:
<Alcaro> huh. why initialize that char buffer? That just looks like a waste of cpu to me.